skrub/_dataframe/tests/test_common.py

```diff
@@ -248,6 +249,33 @@ def test_reset_index(df_module):
     assert col1.index.tolist() == idx
 
 
+@pytest.mark.parametrize("source_is_df", [False, True])
+@pytest.mark.parametrize("target_is_df", [False, True])
+def test_copy_index(source_is_df, target_is_df):
+    source = pd.Series(list("abc"), index=[10, 20, 30], name="s")
+    if source_is_df:
+        source = pd.DataFrame({"s": source}, index=source.index)
+    target = pd.Series(list("def"), index=[100, 200, 300], name="t")
+    if target_is_df:
+        target = pd.DataFrame({"t": target}, index=target.index)
+    out = ns.copy_index(source, target)
+    assert (out.index == source.index).all()
+    assert target.index.tolist() == [100, 200, 300]
+    assert source.index.tolist() == [10, 20, 30]
+
+
+def test_copy_index_non_pandas(all_dataframe_modules):
+    a = all_dataframe_modules["pandas-numpy-dtypes"].example_column
+    b = []
+    assert ns.copy_index(a, b) is b
+    assert ns.copy_index(b, a) is a
+    if "polars" not in all_dataframe_modules:
+        pytest.skip()
```

Comment:
It might be worth to just add the `reason` parameter.

Reply:
added it. I think it might be useful to add fixtures to yield polars and pandas separately so this kind of skip is done in one place only (the fixture)